### PR TITLE
Pull request class refactor

### LINKED ISSUE #254: PullRequest class refactor
Extend `PullRequest` class of methods.

- [x] move PR-related functions from `Project` to `PullRequest`
  - [x] `BasePullRequest`
  - [x] Github
  - [x] Gitlab
  - [x] Pagure
- [x] make `Project`'s methods backward-compatible by calling new methods
- [x] deprecation
  - [x] move deprecated functions to base class
  - [x] deprecate functions on `Project`
- [x] properties
- [x] fix tests

Part of #86
Blocked by #121

### PR DESCRIPTION
**Pull request by mfocko**
Closes #254 

- [x] move PR-related functions from `Project` to `PullRequest`
  - [x] `BasePullRequest`
  - [x] Github
  - [x] Gitlab
  - [x] Pagure
- [x] make `Project`'s methods backward-compatible by calling new methods
- [x] deprecation
  - [x] move deprecated functions to base class
  - [x] deprecate functions on `Project`
- [x] properties
- [x] fix tests
- [x] static methods for `get`, `create`, `list`
- [x] update comments (after refactor from Issue has been merged)
- [x] check docstrings
- [x] `pr_create` -> update name

### CODE REVIEW

File: ogr/abstract.py [hunk 1]
```diff
@@ -270,6 +292,151 @@ def __str__(self) -> str:
...
+    ) -> "PullRequest":
+        """
+        Create a new pull request.
+
+        :param project: Any
+        :param title: str
+        :param body: str
+        :param target_branch: str
+        :param source_branch: str
+        :param fork_username: str The username of forked repository
```
Comment (lachmanfrantisek) [inline ogr/abstract.py]:
```suggestion
        :param fork_username: str The username/namespace of the forked repository.
```

File: ogr/services/gitlab/pull_request.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        self._raw_pr.save()
+        return self
+
+    def merge(self) -> "PullRequest":
+        self._raw_pr.merge()
+        return self
+
+    def add_label(self, *labels: str) -> None:
+        for label in labels:
+            self._raw_pr.labels.append(label)
```
Comment (lachmanfrantisek) [inline ogr/services/gitlab/pull_request.py]: I am wondering if this can work:

```python
self._raw_pr.labels += labels
```

**Comment by mfocko** [comment]
@lachmanfrantisek could you please regenerate the update_pr_info test?
I've had my fun "hacking it", didn't work out. You can squash/amend it into the fix tests commit, should have only yamls in it and constants changed in tests. Thanks.

**Comment by lachmanfrantisek** [comment]
recheck

**Pull request merged**